### PR TITLE
Python: added embedding dimensions support

### PR DESCRIPTION
**Pull request by sinyubonnie-ho**
### Description

 added embedding dimensions support (issue: https://github.com/microsoft/semantic-kernel/issues/5882)

### Contribution Checklist

- [] The code builds clean without any errors or warnings
- [] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [] All unit tests pass, and I have added new tests where possible
- [] I didn't break anyone :smile:

### CODE REVIEW

**Comment by markwallace-microsoft** [comment]
<a href="https://github.com/microsoft/semantic-kernel/blob/06e69aa5bbfe4a00a77ebbb6922955855e035b12/README.md">Py3.10 Test Coverage</a><details><summary>Python 3.10 Test Coverage Report • </summary><table><tr><th>File</th><th>Stmts</th><th>Miss</th><th>Cover</th><th>Missing</th></tr><tbody><tr><td><b>TOTAL</b></td><td><b>5864</b></td><td><b>1061</b></td><td><b>82%</b></td><td>&nbsp;</td></tr></tbody></table><i>report-only-changed-files is enabled. No files were changed during this commit :)</i></details>

## Python 3.10 Unit Test Overview
| Tests | Skipped | Failures | Errors | Time |
| ----- | ------- | -------- | -------- | ------------------ |
| 1257 | 1 :zzz: | 0 :x: | 0 :fire: | 14.810s :stopwatch: |

File: python/semantic_kernel/connectors/ai/open_ai/prompt_execution_settings/open_ai_prompt_execution_settings.py [hunk 1]
```diff
@@ -82,3 +82,4 @@ class OpenAIEmbeddingPromptExecutionSettings(PromptExecutionSettings):
     extra_query: Optional[Dict] = None
     extra_body: Optional[Dict] = None
     timeout: Optional[float] = None
+    dimensions: Optional[int] = None
```
Comment (eavanvalkenburg) [inline python/semantic_kernel/connectors/ai/open_ai/prompt_execution_settings/open_ai_prompt_execution_settings.py:85]: I would like the min and max to be specified using Field, see this: https://docs.pydantic.dev/latest/concepts/fields/#numeric-constraints

**Review by eavanvalkenburg — CHANGES_REQUESTED** [review]

**Comment by eavanvalkenburg** [comment]
Also some ruff and black errors, have a look! @sinyubonnie-ho

File: python/semantic_kernel/connectors/ai/open_ai/prompt_execution_settings/open_ai_prompt_execution_settings.py (hunk 1 above)
Comment (sinyubonnie-ho) [reply python/semantic_kernel/connectors/ai/open_ai/prompt_execution_settings/open_ai_prompt_execution_settings.py]: added numeric constraint. Please check :) Thanks!

**Review by moonbox3 — APPROVED** [review]

**Review by eavanvalkenburg — APPROVED** [review]

**Pull request merged**